deploy/charts/cert-manager/templates/webhook-service.yaml

```diff
@@ -11,6 +11,11 @@ metadata:
     app.kubernetes.io/component: "webhook"
     helm.sh/chart: {{ include "webhook.chart" . }}
 spec:
+  # Allow the K8S API server to reach the webhook server even before its
+  # readiness probes are passing.
+  # This allows the /startupz code in the webhook server to verify that the K8S
+  # API server can connect to it, by performing a dry-run create Certificate.
+  publishNotReadyAddresses: true
```

Comment:
By setting this to true, requests will be routed to unready instances, effectively defeating the point of the check/making things _more_ brittle. I don't think we _can_ safely implement this sort of startupProbe and because of that I don't think we should accept this change.

If we want to make it easier to check readiness of the whole chart, we _could_ add an optional `post-deploy` hook to the Helm chart or similar for verifying this stuff? (or otherwise adding this as part of the `cmctl install` command being added)

Reply:
Even if we do decide to merge this, we _can't_ start defaulting this to `true` by default. It will cause issues during rolling upgrades of the webhook where some number of requests to apiservers will fail (and the apiserver won't perform retries itself, users will have to)